src/RouterCLILookupCodegenBuilder.hack

```diff
@@ -72,11 +72,20 @@ final class RouterCLILookupCodegenBuilder {
       ->setShebangLine('#!/usr/bin/env hhvm')
       ->setFileType(CodegenFileType::HACK_PARTIAL)
       ->setGeneratedFrom($this->generatedFrom)
-      ->setPseudoMainHeader($this->getInitCode())
       ->addClass($this->getCodegenClass($router_classname, $utility_classname))
-      ->setPseudoMainFooterf(
-        '(new %s())->main($argv);',
-        $utility_classname,
+      ->addFunction(
+        $this->cg->codegenFunction('main')
+          ->addEmptyUserAttribute('__EntryPoint')
+          ->setReturnType('void')
+          ->setBodyf(
+            "%s\n".
+            '$argv = '.
+            '\Facebook\TypeAssert\matches<KeyedContainer<int, string>>('.
```

Comment:
hmmm, does this actually work/typecheck? I kinda assumed it wouldn't (as it doesn't with `as`, but maybe reify is a bit more permissive

Reply:
I didn't test it manually but the test passes, which seems to run both the typechecker and the script.

Reply:
Problem with `as` is you can't specify the generic params (`as KeyedContainer<_, _>`) so you get errors about that.